plugins/easyimage/plugin.js

```diff
@@ -239,15 +245,81 @@
 					loader.upload( widgetDef.uploadUrl, widgetDef.additionalRequestParameters );
 
 					fileTools.markElement( img, 'uploadeasyimage', loader.id );
-
-					fileTools.bindNotifications( editor, loader );
 				}
 			}
 
 			data.dataValue = temp.getHtml();
 		} );
 	}
 
+	// Extends given uploadWidget `definition` with an upload progress bar, added within wrapper.
+	function addUploadProgressBar( editor, definition ) {
+		definition.skipNotifications = true;
+		definition.parts.loader = '.cke_loader';
+
+		/*
+		 * Creates a progress bar in a given widget.
+		 *
+		 * Also puts it in it's {@link CKEDITOR.plugins.widget#parts} structure as `progressBar`
+		 *
+		 * @private
+		 * @param {CKEDITOR.plugins.widget} widget
+		 */
+		definition._createProgressBar = function( widget ) {
+			widget.parts.progressBar = CKEDITOR.dom.element.createFromHtml( '<div class="cke_loader">' +
```

Comment:
I wonder if we shouldn't add some ARIA roles for progressbar (like, uhm, `progressbar`).

Reply:
A11y will be provided in a followup issue, we should extract it to a separate story (I mean complete a11y, not limited to progress bar).